packages/next/src/server/use-cache/use-cache-wrapper.ts

```diff
@@ -842,3 +882,21 @@ function createLazyResult<TResult>(
     },
   }
 }
+
+function isPageComponent(
+  args: any[]
+): args is [UseCachePageComponentProps, undefined] {
+  if (args.length !== 2) {
+    return false
+  }
+
+  const [props, ref] = args
+
+  return (
+    ref === undefined && // server components receive an undefined ref arg
+    typeof props === 'object' &&
```

Comment:
need to check not null

Reply:
or we remove the `params` and `searchParams` check completely here, and only rely on the presence of `$$isPageComponent`?

Reply:
well still need to check that it's not null otherwise the property read will throw

Reply:
oops, in my mind I read the comment for the line below this one 🤦‍♂️